src/cc_catalog_airflow/dags/provider_api_scripts/phylopic.py

```diff
@@ -0,0 +1,316 @@
+
+Output:                 TSV file containing the image,
+                        their respective meta-data.
+
+Notes:                  http://phylopic.org/api/
+                        No rate limit specified.
+"""
+
+
+import argparse
+from datetime import datetime, timedelta
+import logging
+
+import common.requester as requester
+import common.storage.image as image
+
+
+logging.basicConfig(
+    format='%(asctime)s - %(name)s - %(levelname)s:  %(message)s',
+    level=logging.INFO
+)
+logger = logging.getLogger(__name__)
+
+
+DELAY = 1
+HOST = 'phylopic.org'
+ENDPOINT = f'http://{HOST}/api/a'
+PROVIDER = 'phylopic'
+LIMIT = 100
+
+
+delayed_requester = requester.DelayedRequester(DELAY)
+image_store = image.ImageStore(provider=PROVIDER)
+
+
+def main(date):
+    """
+    This script pulls the data for a given date from the PhyloPic
+    API, and writes it into a .TSV file to be eventually read
+    into our DB.
+
+    Required Arguments:
+
+    date:  Date String in the form YYYY-MM-DD.  This is the date for
+           which running the script will pull data.
+    """
+
+    param = None
+    offset = 0
+
+    logger.info('Begin: PhyloPic API requests')
+
+    if date == 'all':
+        logger.info('Processing all images')
+        param = {'offset': offset}
+
+        image_count = _get_total_images()
+        logger.info('Total images: {}'.format(image_count))
+
+        while offset <= image_count:
+            _add_data_to_buffer(**param)
+            offset += LIMIT
+            param = {'offset': offset}
+
+    else:
+        param = {'date': date}
+        logger.info('Processing date: {}'.format(date))
+        _add_data_to_buffer(**param)
+
+    image_store.commit()
+
+    logger.info('Terminated!')
+
+
+def _add_data_to_buffer(**args):
+    endpoint = _create_endpoint_for_IDs(**args)
+    IDs = _get_image_IDs(endpoint)
+
+    for id in IDs:
+        if id is not None:
+            details = _get_meta_data(id)
+
+            if details is not None:
+                image_store.add_item(foreign_landing_url=details[1],
+                                     image_url=details[2],
+                                     thumbnail_url=details[3],
+                                     license_url=details[6],
+                                     width=details[4],
+                                     height=details[5],
+                                     creator=details[7],
+                                     title=details[8],
+                                     meta_data=details[9],
+                                     )
+
+
+def _get_total_images():
+    # Get the total number of PhyloPic images
+    total = 0
+    endpoint = 'http://phylopic.org/api/a/image/count'
+    result = _get_response_json(endpoint=endpoint, retries=2)
+
+    if result and result.get('success') is True:
+        total = result.get('result')
+
+    return total if (total is not None) else 0
```

Comment:
```suggestion
        total = result.get('result', 0)

    return total
```
This accomplishes the same, but is more readable to me.  What do you think?